From d56b44d2dfa093883a5c4e91be3f72d99946b170 Mon Sep 17 00:00:00 2001
From: Damien Miller <djm@mindrot.org>
Date: Tue, 04 Feb 2014 00:26:04 +0000
Subject:    - djm@cvs.openbsd.org 2014/02/04 00:24:29

     [ssh.c]
     delay lowercasing of hostname until right before hostname
     canonicalisation to unbreak case-sensitive matching of ssh_config;
     reported by Ike Devolder; ok markus@
---
diff --git a/ssh.c b/ssh.c
index ec95733..add760c 100644
--- a/ssh.c
+++ b/ssh.c
@@ -780,7 +780,6 @@ main(int ac, char **av)
 	if (!host)
 		usage();
 
-	lowercase(host);
 	host_arg = xstrdup(host);
 
 	OpenSSL_add_all_algorithms();
@@ -914,6 +913,7 @@ main(int ac, char **av)
 	}
 
 	/* If canonicalization requested then try to apply it */
+	lowercase(host);
 	if (options.canonicalize_hostname != SSH_CANONICALISE_NO)
 		addrs = resolve_canonicalize(&host, options.port);
 	/*
--
cgit v0.9.2
